app/jobs/pdf_generator.rb

```diff
@@ -0,0 +1,103 @@
+require 'json'
+require 'net/http'
+require 'rmagick'
+
+class PdfGenerator
+    @queue = :ingests
+
+    def self.perform(pid)
+      manifest_url = self.config['vufind_test_url'].to_s + "/Item/" + pid + "/Manifest"
+      json = self.do_get manifest_url
+      parsed_json = JSON.parse json
+      large_jpegs = parsed_json["sequences"][0]["canvases"].map do |current|
+        current["images"][0]["resource"]["@id"]
+      end
+      pdf = self.generate_pdf(large_jpegs)
+      self.add_pdf_to_pid(pdf,pid)
+    end
+
+    def self.add_pdf_to_pid(pdf,pid)
+      resource = Fedora3Object.from_pid pid
+      document_list = Fedora3Object.from_next_pid
+      document_list.parent_pid = resource.pid
+      document_list.model_type = "ListCollection"
+      document_list.title = "Document List"
+      document_list.core_ingest("I")
+      document_list.collection_ingest
+      document_list.list_collection_ingest
+      image_data = self.build_document document_list, pdf, 1
+      self.add_datastreams_to_document pdf, image_data
+    end
+
+    def self.build_document(document_list, document, number)
+      document_data = Fedora3Object.from_next_pid
+      document_data.logger = @logger
+      document_data.parent_pid = document_list.pid
+      document_data.model_type = 'PDFData'
+      document_data.title = "PDF"
+      document_data.core_ingest('I')
+      document_data.data_ingest
+      document_data.document_data_ingest
+      document_data.add_sequence_relationship document_list.pid, number
+      document_data
+    end
+
+    def self.add_datastreams_to_document(document, document_data)
+      document_data.add_datastream_from_file document, 'MASTER', 'application/pdf'
+      document_data.add_master_metadata_datastream
+    end
+
+    def self.do_get(uri)
```

Comment:
I'm pretty sure there's a way we could put some of these commonly used methods like do_get and check_http_error into a Ruby module, and then compose them into the code so we don't need to copy-paste them. I can live with this as-is for the moment, but if we find ourselves having to do this more than once, we should figure that out so the code is more maintainable.